scripts/cat/cats.py

```diff
@@ -1559,39 +1568,220 @@ def thoughts(self):
         i = 0
         # for cats inside the clan
         if where_kitty == "inside":
-            while (
-                other_cat == self.ID
-                and len(all_cats) > 1
-                or (all_cats.get(other_cat).dead and dead_chance != 1)
-                or (other_cat not in self.relationships)
-            ):
-                other_cat = choice(list(all_cats.keys()))
-                i += 1
-                if i > 100:
-                    other_cat = None
-                    break
-        # for dead cats
+            # Roll relation and kin thought chances
+            if thought_type == 'relationship':
```

Comment:
The thought type should be handled before the "if where_kitty == " statement. In that way there won't be that many code duplications in the class.